### ISSUE TITLE
[ui5-tooling-transpile] source maps loose relationship to `.ts` files with minify task

### ISSUE THREAD

**Issue by Kasde96** [ui5-community/ui5-ecosystem-showcase#790]
Hello there, 

I want to debug my ts files in an on prem system. This doesn't work because the js.map files in the source property point to the dbg.js files and not the ts files. Therefore, the ts files never show up in the development console in the Fiori Launchpad.

@petermuessig asked me to open a ticket here. His workaround is to --exclude-task minify in the build operation.
[https://github.com/SAP/open-ux-tools/issues/1162](url)

Could you please fix this?

Thanks

**Comment by Kasde96** [comment]
@petermuessig 
I think it would be better if the -dbg.js files would have a map.js-File that point to the ts-Sources. Would this be possible?

**Comment by petermuessig** [comment]
I'll start a discussion with @codeworrior, @RandomByte, and @matz3 regarding the UI5 Tooling. The `minify` task isn't considering transpiled sources so far having sourcemaps. I need to understand the flow right now. Maybe I can create a workaround in the `ui5-tooling-transpile` - but I need to understand the issue a bit more in detail and find a proper way how to overcome / to fix the loss of the sourcemap link.

**Comment by Kasde96** [comment]
Yes i think it would be the best decission to fix it in the `ui5-tooling-transpile`. Do you need any detailed informations from me?

**Comment by RandomByte** [comment]
We are already tracking this team internally (CPOUI5FOUNDATION-467). A while ago I also created this PoC to handle existing "inline" source maps correctly: https://github.com/SAP/ui5-builder/pull/780

**Comment by petermuessig** [comment]
I added a suggestion to the PR mentioned above to make it more complete: https://github.com/SAP/ui5-builder/pull/780#issuecomment-1675807428

**Comment by petermuessig** [comment]
Still waiting for final submit of Draft PR https://github.com/SAP/ui5-builder/pull/780 - due to vacation season the UI5 tooling team is a bit delayed. But change is in a good shape already... Let's give them some more days...

**Comment by RandomByte** [comment]
Hey sorry for the delay but we were facing quite some issues in our internal integration tests.

We were able to resolve them and the feature for supporting input source maps is now released as part of UI5 CLI v3.7.0.

Documentation should follow soon. There are some caveats which shouldn't be relevant for the typical scenarios where the ecosystem-showcase transpile task is being used. 

The transpile task creates source maps and references them in the transpiled JavaScript resources. The updated minify task will now find these references and use the respective source maps during minification (done by Terser). Ultimately the minified resource will then have a source map that maps back to the original TypeScript source 👍🏼
